man/systemd.pcrlock.xml

```diff
@@ -0,0 +1,298 @@
+        firmware control transitions into boot loader/OS control. Statically defined.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>500-separator.pcrlock</filename></term>
+
+        <listitem><para>The EFI "separator" measurements on PCRs 0-6 done once by the firmware to indicate
+        where firmware control transitions into boot loader/OS control. Statically defined.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>550-firmware-code-late.pcrlock</filename></term>
+
+        <listitem><para>Firmware code measurements, as recorded to PCR 0 and 2, after the separator
+        measurement (see <filename>400-secureboot-separator.pcrlock.</filename> above). May be generated via
+        <command>systemd-pcrlock lock-firmware-code</command>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>550-firmware-config-late.pcrlock</filename></term>
+
+        <listitem><para>Firmware configuration measurements, as recorded to PCR 1 and 3, after the separator
+        measurement (see <filename>400-secureboot-separator.pcrlock.</filename> above). May be generated via
+        <command>systemd-pcrlock lock-firmware-config</command>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>600-gpt.pcrlock</filename></term>
+
+        <listitem><para>The GPT partition table of the booted medium, as recorded to PCR 5 by the
+        firmware. May be generated via <command>systemd-pcrlock lock-gpt</command>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>620-secureboot-authority.pcrlock</filename></term>
+
+        <listitem><para>The SecureBoot authority, as recorded to PCR 7. May be generated via
+        <command>systemd-pcrlock lock-secureboot-authority</command>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>700-action-efi-exit-boot-services.pcrlock</filename></term>
+
+        <listitem><para>The EFI action generated when <function>ExitBootServices()</function> is generated,
+        i.e. the UEFI environment is left and the OS takes over. Covers the PCR 5 measurement. Statically
+        defined.</para>
+
+        <xi:include href="version-info.xml" xpointer="v255"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><filename>710-kernel-cmdline.pcrlock</filename></term>
+
+        <listitem><para>The kernel command line, as measured by the Linux kernel to PCR 9. May be generated
```

Comment:
According to the man pages, `systemd-boot` and `systemd-stub` measures the kernel command line to PCR 12. Is this a typo or intentional?

Reply:
Intentional. Recent kernel versions started to measure the kernel command line on their own into PCR12. 

So yeah, the kernel command line might be measured thrice these days:

1. once by sd-boot into PCR 9
2. once by sd-stub into PCR 9
3. once by the kernel itself into PCR 12

you can never have enough measurements...

btw, I changed sd-boot recently to not measure the cmdline of UKIs unless it was modified from the default as stored in the UKI, since it is then covered by the UKI. This measn that the first measurement should not happen anymore if sd-boot + sd-stub are combined.